development-tools/clang.rst

```diff
@@ -49,177 +54,94 @@ A complete list of sanitizers can be found at `Controlling Code Generation
-    libclang_rt.dfsan-x86_64.a  libclang_rt.san-x86_64.a
-    libclang_rt.full-x86_64.a   libclang_rt.tsan-x86_64.a
-    libclang_rt.lsan-x86_64.a   libclang_rt.ubsan_cxx-x86_64.a
-    libclang_rt.msan-x86_64.a   libclang_rt.ubsan-x86_64.a
+Python build setup
+==================
 
-On macOS, the libraries are installed in
-``/usr/local/lib/clang/3.3/lib/darwin/``:
+This portion of the document covers invoking Clang and LLVM with the options
+required so the sanitizers analyze Python with under its test suite.
 
-.. code-block:: console
+Set the compiler to Clang, in case it's not the default::
 
-    $ ls /usr/local/lib/clang/3.3/lib/darwin/
-    libclang_rt.10.4.a                    libclang_rt.ios.a
-    libclang_rt.asan_osx.a                libclang_rt.osx.a
-    libclang_rt.asan_osx_dynamic.dylib    libclang_rt.profile_ios.a
-    libclang_rt.cc_kext.a                 libclang_rt.profile_osx.a
-    libclang_rt.cc_kext_ios5.a            libclang_rt.ubsan_osx.a
-    libclang_rt.eprintf.a
+   export CC="clang"
 
-.. note::
+If you want to use additional sanitizer options (found in Clang documentation),
+add them to the ``CFLAGS`` variable.
+For example, you may want the checked process to exit after the first failure::
 
-    You should never have to add the libraries to a project. Clang will handle
-    it for you. If you find you cannot pass the ``-fsanitize=XXX`` flag through
-    ``make``'s implicit variables (``CFLAGS``, ``CXXFLAGS``, ``CC``,
-    ``CXXFLAGS``, ``LDFLAGS``) during ``configure``, then you should modify the
-    makefile after configuring to ensure the flag is passed through the
-    compiler.
+   export CFLAGS="-fno-sanitize-recover"
 
-The installer does not install all the components needed on occasion. For
-example, you might want to run a ``scan-build`` or examine the results with
-``scan-view``. You can copy the components by hand with: ::
+Then, run ``./configure`` with the relevant flags:
 
-    sudo mkdir /usr/local/bin/scan-build
-    sudo cp -r llvm-3.4/tools/clang/tools/scan-build /usr/local/bin
-    sudo mkdir /usr/local/bin/scan-view
-    sudo cp -r llvm-3.4/tools/clang/tools/scan-view /usr/local/bin
+* ASan: ``--with-address-sanitizer --without-pymalloc``
+* UBsan: ``--with-undefined-behavior-sanitizer``
 
-.. note::
+It is OK to specify both sanitizers.
 
-    Because the installer does not install all the components needed on
-    occasion, you should not delete the scratch directory until you are sure
-    things work as expected. If a library is missing, then you should search for
-    it in the Clang/LLVM build directory.
+After that, run ``make`` and ``make test`` as usual.
+Note that ``make`` itself may fail.
 
-Python build setup
-==================
 
-This portion of the document covers invoking Clang and LLVM with the options
-required so the sanitizers analyze Python with under its test suite. Two
-checkers are used - ASan and UBSan.
+Thorough build setup
```

Comment:
```suggestion
Build setup for sanitizers
```

Reply:
Thank you for the review!

I see this is unclear. Instead of "Thorough build" I switched to "enabling sanitizers for all code", and reworded the section a bit more.